ftfy/bad_codecs/utf8_variants.py

```diff
@@ -46,8 +46,29 @@
 import codecs
 
 NAME = 'utf-8-variants'
-# This regular expression matches all possible six-byte CESU-8 sequences.
-CESU8_RE = re.compile(b'\xed[\xa0-\xaf][\x80-\xbf]\xed[\xb0-\xbf][\x80-\xbf]')
+
+# This regular expression matches all possible six-byte CESU-8 sequences,
+# plus truncations of them at the end of the string.
+CESU8_EXPR = (
+    b'(\xed[\xa0-\xaf][\x80-\xbf]\xed[\xb0-\xbf][\x80-\xbf]'
+    b'|\xed([\xa0-\xaf]([\x80-\xbf](\xed[\xb0-\xbf]?)?)?)?$'
+    b')'
+)
```

Comment:
Huh, I actually find that more confusing. The idea, I suppose, is that once you've matched a `$`, you can't possibly match anything else but `$`?